pkg/services/ngalert/schedule/schedule.go

```diff
@@ -363,15 +370,42 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 			if results == nil {
 				results = append(results, eval.NewResultFromError(err, e.scheduledAt, dur))
 			}
+			if err == nil {
+				for _, result := range results {
+					if result.Error != nil {
+						err = multierror.Append(err, result.Error)
+					}
+				}
+			}
+			span.RecordError(err)
```

Comment:
If we pass `nil` to `RecordError` is it a no-op?

Reply:
nil is not possible. That branch is executed when either `err` is not nil or `results` have errors, i.e. there is at least one result that contains error and therefore `err` will be assigned.